activerecord/test/cases/associations/has_one_associations_test.rb

```diff
@@ -828,6 +828,18 @@ class DestroyByParentAuthor < ActiveRecord::Base
     assert_not DestroyByParentBook.exists?(book.id)
   end
 
+  test "does not destroy associated when building new record with primary_key manually set" do
+    old_author = DestroyByParentAuthor.create!(name: "Test")
+    old_book = DestroyByParentBook.create!(author: old_author)
+
+    DestroyByParentAuthor.new(
+      id: old_author.id,
+      book: DestroyByParentBook.new
+    )
```

Comment:
I think we should assert no queries are made at all:
```suggestion
    assert_no_queries do
      DestroyByParentAuthor.new(
        id: old_author.id,
        book: DestroyByParentBook.new
      )
    end
```

Reply:
Sadly the only way (at least for now) of asserting no queries is using: 
`return self.target = record unless owner.persisted?` before `return target unless load_target || record` on the line 57
line 57 executes a SELECT query
So using this assert would prevent me of following this recommendation https://github.com/rails/rails/pull/42137#discussion_r641959159